src/actions/guides/http_and_routing/routing_and_params.cr

```diff
@@ -336,7 +336,99 @@ class Guides::HttpAndRouting::RoutingAndParams < GuideAction
 
     > Learn more about [error handling](#{Guides::HttpAndRouting::ErrorHandling.path}).
 
-    ## Query parameters
+    ## Handling parameters
+
+    Parameters, or `params`, are data that is sent from client back to the server. There are a few different ways this can happen:
+
+    * Path parameters - The dynamic values passed in your route path. e.g. `/users/:id`.
+    * Query parameters - These are added to the end of a URL after the `?` in key/value pairs. e.g. `?page=1`
+    * Form parameters - When you perform an HTTP POST, the data sent through the body. This may be formatted as JSON, or form data
+    * Multipart parameters - Similar to form parameters, but generally to contain a file
+
+    When parameters are passed to an action, you have access to these with the `params` method.
+
+    ```crystal
+    # src/actions/users/index.cr
+    class Users::Index < BrowserAction
+
+      # example.com/users?page=1&filter=active
+      get "/users" do
+        # These are both type `String`
+        filter = params.get(:filter)
+        page = params.get(:page)
+
+        # This is missing and will raise an error
+        per = params.get(:per)
+        plain_text "All users starting on page \#{page}"
+      end
+    end
+    ```
+
+    You can also use the `params.get?(:key)` method to return `nil` if the key doesn't exist.
+
+    ```crystal
+    # example.com/users?page=1&filter=active
+    get "/users" do
+      # This will return nil
+      per = params.get?(:per)
+      plain_text "..."
+    end
+    ```
+
+    > By default, all param values are trimmed of blankspace. If you need the raw value,
+    > use `params.get_raw(:key)` or `params.get_raw?(:key)`.
+
+    ### Params from query string
+
+    Returns just the query params as `HTTP::Params`.
+
+    You can use the `params.get(:key)` method to get your values from the query string,
```

Comment:
It feels odd that this restates what you just did above. Maybe you could say something like "As seen above..." or maybe it doesn't matter because no one reads docs like a book 😆

Reply:
Yeah, my thought was if someone on the home page searches "params", then lands on that section, they might not read above it. It is a bit redundant, but maybe it's ok since most people jump around in docs?